src/api/models/search/vnQueryOptions.ts

```diff
@@ -0,0 +1,47 @@
+import type { Operator } from '../operator';
+import type { BaseFilter } from './baseFilter';
+import type { BaseQueryOptions } from '../baseQueryOptions';
+
+/** List of available sort fields for vn. */
+export const VN_SORT_FIELDS = ['id', 'title', 'released', 'popularity', 'rating', 'votecount'] as const;
+
+/** Represents sort field for vn. */
+export type VnSortField = typeof VN_SORT_FIELDS[number];
+
+/** List of search fields for vn. */
+export const VN_SEARCH_FIELDS = ['search', 'lang', 'olang', 'platform', 'tag'] as const;
+
+/** Represents available search field for vn. */
+export type VnSearchField = typeof VN_SEARCH_FIELDS[number];
+
+/** Represents available filters for vn. */
+export type VnFilter = BaseFilter<VnSearchField, Extract<Operator, '=' | '!='>> | BaseFilter<VnSortField, Operator>;
```

Comment:
I don't think we benefit from `Extract` here, because it allows providing type that don't extend `Operator`. 
I suggest adding string extract
```ts
type ExtractStrict<T, K extends T> = Extract<T, K>
```